compiler/src/dmd/mtype.d

```diff
@@ -2386,13 +2386,15 @@ extern (C++) abstract class Type : ASTNode
     /**************************
      * Return type with the top level of it being mutable.
      */
-    inout(Type) toHeadMutable() inout
+    inout(Type) toHeadMutable() inout nothrow
     {
         if (!mod)
             return this;
         Type unqualThis = cast(Type) this;
         // `mutableOf` needs a mutable `this` only for caching
-        return cast(inout(Type)) unqualThis.mutableOf();
+        try
+            return cast(inout(Type)) unqualThis.mutableOf();
+        catch (Throwable) assert(0);
```

Comment:
again

Reply:
Putting it another way, `nothrow` is a promise that no `Throwables` are thrown. It is not a promise about `Exceptions`.